docs/framework/interop/how-to-map-hresults-and-exceptions.md

```diff
@@ -15,127 +15,112 @@ ms.assetid: 610b364b-2761-429d-9c4a-afbc3e66f1b9
-|**All other HRESULTs**|**COMException**|  
-  
- To retrieve extended error information, the managed client must examine the fields of the exception object that was generated. For the exception object to provide useful information about an error, the COM object must implement the **IErrorInfo** interface. The runtime uses the information provided by **IErrorInfo** to initialize the exception object.  
-  
- If the COM object does not support **IErrorInfo**, the runtime initializes an exception object with default values. The following table lists each field associated with an exception object and identifies the source of default information when the COM object supports **IErrorInfo**.  
-  
- Note that the runtime will sometimes ignore an `HRESULT` in cases where there is an `IErrorInfo` present on the thread.  This behavior can occur in cases where the `HRESULT` and the `IErrorInfo` do not represent the same error.  
-  
-|Exception field|Source of Information from COM|  
-|---------------------|------------------------------------|  
-|**ErrorCode**|HRESULT returned from call.|  
-|**HelpLink**|If **IErrorInfo->HelpContext** is nonzero, the string is formed by concatenating **IErrorInfo->GetHelpFile** and "#" and **IErrorInfo->GetHelpContext**. Otherwise the string is returned from **IErrorInfo->GetHelpFile**.|  
-|**InnerException**|Always a null reference (**Nothing** in Visual Basic).|  
-|**Message**|String returned from **IErrorInfo->GetDescription**.|  
-|**Source**|String returned from **IErrorInfo->GetSource**.|  
-|**StackTrace**|The stack trace.|  
-|**TargetSite**|The name of the method that returned the failing HRESULT.|  
-  
- Exception fields, such as **Message**, **Source**, and **StackTrace** are not available for the **StackOverflowException**.  
-  
+    }
+    }
+    CMyClass::MethodThatThrows
+    {
+    throw new NoAccessException();
+    }
+    ```
+
+ You might encounter a program (in any programming language) that uses both managed and unmanaged code at the same time. For example, the custom marshaler in the following code example uses the `Marshal.ThrowExceptionForHR(int HResult)` method to throw an exception with a specific HRESULT value. The method looks up the HRESULT and generates the appropriate exception type. For example, the HRESULT in the following code fragment generates `ArgumentException`.
+
+```cpp
+CMyClass::MethodThatThrows
+{
+    Marshal.ThrowExceptionForHR(COR_E_ARGUMENT);
+}
+```
+
+ The following table provides the common mappings from HRESULT to its comparable exception class in .NET. HRESULT values without explicit mappings are mapped to `COMException`. The complete up-to-date mapping can be found in the [dotnet/runtime repository](https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/rexcep.h).
+
+|HRESULT|.NET exception|
+|-------------|--------------------|
+|`COR_E_APPLICATION`|`ApplicationException`|
+|`COR_E_ARGUMENT or E_INVALIDARG`|`ArgumentException`|
```

Comment:
The "or" shouldn't be included in the code format.  That would prevent it from being localized.
```suggestion
|`COR_E_ARGUMENT` or `E_INVALIDARG`|`ArgumentException`|
```